> > Currently, tests fail with:
> > ```
> > Executing pytestCheckPhase
> > ERROR: usage: __main__.py [options] [file_or_dir] [file_or_dir] [...]
> > __main__.py: error: unrecognized arguments: -n=auto
> >   inifile: /build/source/pyproject.toml
> >   rootdir: /build/source
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Edit related to https://github.com/jrnl-org/jrnl/blob/v3.0/pyproject.toml#L133
> > Maybe pytest version? https://github.com/jrnl-org/jrnl/blob/v3.0/pyproject.toml#L161
> 
> Do you really need to create a override? Doesn't relaxing deps works?

I do not know much about relaxing deps, but I think that's not the case, with more recent versions of pytest-bdd there's a few errors that broke half of the test suite:

```
...
_ test_count_modifications_when_editing_whole_journal_and_adding_to_last_entry[basic_dayone.yaml] _
[gw7] linux -- Python 3.10.5 /nix/store/9wa02q541sxq7372f8zv27rl57aribxj-python3-3.10.5/bin/python3.10
Traceback (most recent call last):
  File "/nix/store/i39ans7sfvqjjlvlqdz7zvrcyl73mhn8-python3.10-pytest-bdd-6.0.0/lib/python3.10/site-packages/pytest_bdd/scenario.py", line 184, in scenario_wrapper
    _execute_scenario(feature, scenario, request)
  File "/nix/store/i39ans7sfvqjjlvlqdz7zvrcyl73mhn8-python3.10-pytest-bdd-6.0.0/lib/python3.10/site-packages/pytest_bdd/scenario.py", line 154, in _execute_scenario
    _execute_step_function(request, scenario, step, step_func)
  File "/nix/store/i39ans7sfvqjjlvlqdz7zvrcyl73mhn8-python3.10-pytest-bdd-6.0.0/lib/python3.10/site-packages/pytest_bdd/scenario.py", line 112, in _execute_step_function
    for arg, value in parser.parse_arguments(step.name).items():
  File "/nix/store/i39ans7sfvqjjlvlqdz7zvrcyl73mhn8-python3.10-pytest-bdd-6.0.0/lib/python3.10/site-packages/pytest_bdd/parsers.py", line 68, in parse_arguments
    return cast(Dict[str, Any], self.parser.parse(name).named)
AttributeError: 'NoneType' object has no attribute 'named'
___ test_correctly_count_when_removing_entries_via_edit[basic_onefile.yaml] ____
[gw0] linux -- Python 3.10.5 /nix/store/9wa02q541sxq7372f8zv27rl57aribxj-python3-3.10.5/bin/python3.10
Traceback (most recent call last):
  File "/nix/store/i39ans7sfvqjjlvlqdz7zvrcyl73mhn8-python3.10-pytest-bdd-6.0.0/lib/python3.10/site-packages/pytest_bdd/scenario.py", line 184, in scenario_wrapper
    _execute_scenario(feature, scenario, request)
  File "/nix/store/i39ans7sfvqjjlvlqdz7zvrcyl73mhn8-python3.10-pytest-bdd-6.0.0/lib/python3.10/site-packages/pytest_bdd/scenario.py", line 154, in _execute_scenario
    _execute_step_function(request, scenario, step, step_func)
  File "/nix/store/i39ans7sfvqjjlvlqdz7zvrcyl73mhn8-python3.10-pytest-bdd-6.0.0/lib/python3.10/site-packages/pytest_bdd/scenario.py", line 112, in _execute_step_function
    for arg, value in parser.parse_arguments(step.name).items():
  File "/nix/store/i39ans7sfvqjjlvlqdz7zvrcyl73mhn8-python3.10-pytest-bdd-6.0.0/lib/python3.10/site-packages/pytest_bdd/parsers.py", line 68, in parse_arguments
    return cast(Dict[str, Any], self.parser.parse(name).named)
AttributeError: 'NoneType' object has no attribute 'named'
_ test_writing_an_entry_at_the_prompt_no_editor_should_store_the_entry[empty_folder.yaml] _
[gw6] linux -- Python 3.10.5 /nix/store/9wa02q541sxq7372f8zv27rl57aribxj-python3-3.10.5/bin/python3.10
Traceback (most recent call last):
  File "/nix/store/i39ans7sfvqjjlvlqdz7zvrcyl73mhn8-python3.10-pytest-bdd-6.0.0/lib/python3.10/site-packages/pytest_bdd/scenario.py", line 184, in scenario_wrapper
    _execute_scenario(feature, scenario, request)
  File "/nix/store/i39ans7sfvqjjlvlqdz7zvrcyl73mhn8-python3.10-pytest-bdd-6.0.0/lib/python3.10/site-packages/pytest_bdd/scenario.py", line 154, in _execute_scenario
    _execute_step_function(request, scenario, step, step_func)
  File "/nix/store/i39ans7sfvqjjlvlqdz7zvrcyl73mhn8-python3.10-pytest-bdd-6.0.0/lib/python3.10/site-packages/pytest_bdd/scenario.py", line 112, in _execute_step_function
    for arg, value in parser.parse_arguments(step.name).items():
  File "/nix/store/i39ans7sfvqjjlvlqdz7zvrcyl73mhn8-python3.10-pytest-bdd-6.0.0/lib/python3.10/site-packages/pytest_bdd/parsers.py", line 68, in parse_arguments
    return cast(Dict[str, Any], self.parser.parse(name).named)
AttributeError: 'NoneType' object has no attribute 'named'
...
``` 

You can check full error log at https://hydra.nixos.org/build/184485353 .

And based on https://github.com/jrnl-org/jrnl/blob/develop/poetry.lock#L732 is the reason that I am trying to override it.

--- a/pkgs/applications/misc/jrnl/default.nix
+++ b/pkgs/applications/misc/jrnl/default.nix
@@ -5,14 +5,14 @@
 
 python3.pkgs.buildPythonApplication rec {
   pname = "jrnl";
-  version = "2.8.4";
+  version = "3.0";
   format = "pyproject";
 
   src = fetchFromGitHub {
     owner = "jrnl-org";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-Edu+GW/D+R5r0R750Z1f8YUVPMYbm9PK4D73sTDzDEc=";
+    sha256 = "sha256-wyN7dlAbQwqvES8qEJ4Zo+fDMM/Lh9tNjf215Ywop10=";
   };
 
   nativeBuildInputs = with python3.pkgs; [
@@ -31,12 +31,22 @@ python3.pkgs.buildPythonApplication rec {
     pyxdg
     pyyaml
     tzlocal
+    ruamel-yaml
+    rich
   ];
 
   checkInputs = with python3.pkgs; [
     pytest-bdd
+    pytest-xdist
     pytestCheckHook
     toml
+  ];
+
+  # Upstream expects a old pytest-bdd version
+  # Once it changes we should update here too
+  # https://github.com/jrnl-org/jrnl/blob/develop/poetry.lock#L732
+  disabledTests = [
+    "bdd"
   ];
 
   postPatch = ''

